### ISSUE TITLE
couch store sessions should expire

### ISSUE THREAD

**Issue by tyler-johnson** [BeneathTheInk/couchdb-jwt-auth-server#16]
The redis store will set a TTL value on sessions so that they can be auto cleaned. CouchDB does not have this out-of-box like redis does, but we we could fake it by setting an expire property on the document and having the store check the value before verifying it.

**Comment by dariocravero** [comment]
I've been thinking about the session store for a while and I'm a bit split with using couchdb as a store and it's exactly because of the TTL/expiration of tokens (well, the lack of). I also think that Redis is best suited for this use case and using it would simplify the logic around the session store greatly. I know it's great to be able to configure stuff but I'm wondering, if storing your sessions in couchdb is a suboptimal choice, why would one allow for it to be used as a store? (apart from the fact that you already have couchdb there) What are your thoughts on this?

**Comment by tyler-johnson** [comment]
> if storing your sessions in couchdb is a suboptimal choice, why would one allow for it to be used as a store?

Yes these were my thoughts too when I was initially created this. Actually the Redis store was the first store I created and used because I knew that's what I would be using in practice and fits the use case so much better.

> (apart from the fact that you already have couchdb there)

And this was the reason I ultimately separated the Redis store and included the CouchDB store. I figured the couch store was good enough for users that wanted to get up and running quickly without worrying about yet-another-database to configure.

**Comment by dariocravero** [comment]
Right but in that case an in memory store should be good enough, shouldn't
it?

**Comment by tyler-johnson** [comment]
Well I figured it wasn't good enough just since it does not scale and everything is deleted when the app is restarted.

However I see your point: if the CouchDB store is also not recommended in production is there any point to including it? The answer may very well be no, something I will have to do more thinking on.
